context/src/main/java/io/grpc/Context.java

```diff
@@ -379,8 +379,13 @@ public Context attach() {
 
   /**
    * Detach the current context and attach the provided replacement which should be the context of
-   * the outer scope, thus exit the current scope.  If this context is not {@link #current()} a
-   * SEVERE message will be logged but the context to attach will still be bound.
+   * the outer scope, thus exit the current scope.
+   *
+   * <p>This context should be {@link #current()}.  If it's not the case, a SEVERE message will be
```

Comment:
I don't think this is the way we want to describe it. The worst thing to is `Context.current().detach()`. Instead, you should say this context should be the attached context (the exact reverse of the previous attach). And then include other details about if the attached context isn't current, then probably some other code is broken and not detaching correctly, but it will be logged here.